sydent/http/servlets/__init__.py

```diff
@@ -163,45 +163,27 @@ def inner(self, request: Request, *args, **kwargs) -> bytes:
 
 
 def asyncjsonwrap(f):
-    def reqDone(resp: Dict[str, Any], request: Request) -> None:
-        """
-        Converts the given response content into JSON and encodes it to bytes, then
-        writes it as the response to the given request with the right headers.
-
-        :param resp: The response content to convert to JSON and encode.
-        :param request: The request to respond to.
-        """
-        request.setHeader("Content-Type", "application/json")
-        request.write(dict_to_json_bytes(resp))
-        request.finish()
-
-    def reqErr(failure: Failure, request: Request) -> None:
-        """
-        Logs the given failure. If the failure is a MatrixRestError, writes a response
-        using the info it contains, otherwise responds with 500 Internal Server Error.
-
-        :param failure: The failure to process.
-        :param request: The request to respond to.
-        """
+    async def render(f, servlet, request, **kwargs):
```

Comment:
We don't need to pass `f` in here since the function has access to it from the parent scope, similar to how `inner` just uses it directly.

Reply:
So I wondered about this--I thought that `f` would be in scope, but I found that using `f` without explicitly passing it in results in an `nameError: 'f' is not defined`, so I passed it explicitly. I am new to decorator functions so I figured there was something about the scope that I didn't understand here. Setting a breakpoint directly under line `166` when `f` is not passed in does indeed show that `f` is undefined. So I'm stumped! Let me know what you think, and I'll hold off on pushing the other changes until we figure out what to do here.

Reply:
Huh, weirdly this seems to work with non-async functions, yet I can't find anything that says scope should be different here.

Anyway, passing through `f` should be fine in that case.

I also think it is worth pointing out how this function now looks similar to `jsonwrap` which is great!

Reply:
Awesome I will push the other changes!

Reply:
Looks good overall! I have a couple minor nits and I think we can merge this:

* Let's rename `servlet` -> `self` (this is a bit odd, but it is common to do and matches the `jsonwrap` version).
* Add a type to `request`.
* Use `functools.wraps` on `inner`.
* Update the docstring for `inner` due to the changes we've done!